paasta_tools/instance/kubernetes.py

```diff
@@ -73,6 +75,44 @@ def set_cr_desired_state(
         raise RuntimeError(error_message)
 
 
+def autoscaling_status(
+    kube_client: kubernetes_tools.KubeClient,
+    job_config: LongRunningServiceConfig,
+    namespace: str,
+):
+    status = {}
+    try:
+        hpa = kube_client.autoscaling.read_namespaced_horizontal_pod_autoscaler(
+            name=job_config.get_sanitised_deployment_name(), namespace=namespace
+        )
```

Comment:
I think only this line might throw ApiException. Could you move the remaining out of the try block?

Reply:
Instead of this, I caught all exceptions and throw a runtime exception. If anything goes wrong, we can pinpoint the cause easily.

Reply:
Is this a good idea? If the api call failed. I think we should make this function fail gracefully rather than throw the entire traceback. Similar to what we did for https://jira.yelpcorp.com/browse/PAASTA-16182

Reply:
Fixed.